Split installation instructions out of quickstart (#1679)
cocotb/cocotb: documentation/source/install.rst

==============

Cocotb has the following requirements:

* Python 3.5+
* Python-dev packages
* GCC 4.8.1+ or Clang 3.3+ and associated development packages
* GNU Make
* A Verilog or VHDL simulator, depending on your RTL source code

.. versionchanged:: 1.4 Dropped Python 2 support

.. note:: In order to use a 32-bit simulator you need to use a 32-bit version of Python.

Linux - Debian/Ubuntu-based Systems
-----------------------------------

In a terminal, run

.. code-block:: bash

    sudo apt-get install make gcc g++ python3 python3-dev python3-pip
    sudo apt-get install swig  # optional, needed for one of the examples

.. seealso::

   For more installation options, please see `our Wiki <https://github.com/cocotb/cocotb/wiki/Tier-2-Setup-Instructions>`_.

Linux - Red Hat-based Systems
-----------------------------

In a terminal, run

.. code-block:: bash

    sudo yum install make gcc gcc-c++ libstdc++-devel python3 python3-devel python3-pip
    sudo yum install swig  # optional, needed for one of the examples

.. seealso::

   For more installation options, please see `our Wiki <https://github.com/cocotb/cocotb/wiki/Tier-2-Setup-Instructions>`_.

Windows
-------

> `Conda <https://conda.io/>`_ is an open-source package and environment management system that we recommend for Windows.

Reviewer: Maybe `conda` should be first (possibly easiest way, also does not need any root access) for all OSes.

************
Installation
************

Using cocotb requires :ref:`installation of pre-requisites<install-pre-requisites>` and
:ref:`installation of cocotb<installation-via-pip>` itself.
In this document, we are assuming that you already have a
:ref:`supported simulator<simulator-support>` available in ``PATH``.

.. _install-pre-requisites:

Pre-requisites
==============

Cocotb has the following requirements:

* Python 3.5+
* Python-dev packages
* GCC 4.8.1+ or Clang 3.3+ and associated development packages
* GNU Make
* A Verilog or VHDL simulator, depending on your RTL source code

.. versionchanged:: 1.4 Dropped Python 2 support

.. note:: In order to use a 32-bit simulator you need to use a 32-bit version of Python.

Windows
-------

`Conda <https://conda.io/>`_ is an open-source package and environment management system that we recommend for Windows.

Download and install `Miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ from https://conda.io/.
From a Command Prompt or the Windows PowerShell, use the following line to install a compiler (GCC or Clang) and GNU Make:

.. code-block:: bash

    conda install -c msys2 m2-base m2-make m2w64-toolchain libpython

> 

Reviewer: ```suggestion

.. note:: 

    In case errors related to ``Microsoft Visual Studio`` you need to set MinGW as the default compiler details can be found `here <https://stackoverflow.com/questions/3297254/how-to-use-mingws-gcc-compiler-when-installing-python-package-using-pip>`_.

```

.. _install:

************
Installation
************

Using cocotb requires installation of prerequisites and installation of cocotb itself.
In this document, we are assuming that you already have a
:ref:`supported simulator<simulator-support>` available in ``PATH``.

.. _install-prerequisites:

Installation of Prerequisites
=============================

Cocotb has the following requirements:

* Python 3.5+
* Python-dev packages
* GCC 4.8.1+ or Clang 3.3+ and associated development packages
* GNU Make
* A Verilog or VHDL simulator, depending on your RTL source code

.. versionchanged:: 1.4 Dropped Python 2 support

.. note:: In order to use a 32-bit simulator you need to use a 32-bit version of Python.

The installation instructions vary depending on your operating system:

.. tabs::

   .. group-tab:: Windows

      `Conda <https://conda.io/>`_ is an open-source package and environment management system that we recommend for Windows.

      Download and install `Miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ from https://conda.io/.

>       From a Command Prompt or the Windows PowerShell, use the following line to install a compiler (GCC or Clang) and GNU Make:

Reviewer: ```suggestion
      From an Anaconda Prompt use the following line to install a compiler (GCC or Clang) and GNU Make:
```

Author: Good point.

> .. _install:

Reviewer 2: Nit: would marginally prefer `installation.rst`. I'll see what other projects do for this first.

Author: Google says ``"source/installation.rst"`` wins with 2.1k vs. 1.6k hits.
But we also have ``install_devel.rst``, should this then be called ``installation_devel.rst`` or ``installation_development.rst``?

(While we are at it, Sphinx' default for the directory would be ``docs/source/``.)

Reviewer 2: Not a big deal - although it is tempting to rename the folder to "docs". Thanks for collecting the data!

.. _install:

************
Installation
************

Using cocotb requires installation of prerequisites and installation of cocotb itself.
In this document, we are assuming that you already have a
:ref:`supported simulator<simulator-support>` available in ``PATH``.

.. _install-prerequisites:

Installation of Prerequisites
=============================

Cocotb has the following requirements:

* Python 3.5+
* Python-dev packages
* GCC 4.8.1+ or Clang 3.3+ and associated development packages
* GNU Make
* A Verilog or VHDL simulator, depending on your RTL source code

.. versionchanged:: 1.4 Dropped Python 2 support

.. note:: In order to use a 32-bit simulator you need to use a 32-bit version of Python.

> The installation instructions vary depending on your operating system.
> Please choose:

Reviewer 3: ```suggestion
The installation instructions vary depending on your operating system:
```

.. versionchanged:: 1.4 Dropped Python 2 support

.. note:: In order to use a 32-bit simulator you need to use a 32-bit version of Python.

The installation instructions vary depending on your operating system:

.. tabs::

   .. group-tab:: Windows

      `Conda <https://conda.io/>`_ is an open-source package and environment management system that we recommend for Windows.

      Download and install `Miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ from https://conda.io/.
      From a Command Prompt or the Windows PowerShell, use the following line to install a compiler (GCC or Clang) and GNU Make:

      .. code-block::

         conda install -c msys2 m2-base m2-make m2w64-toolchain libpython

      .. container:: toggle

          .. container:: header

              **Show/Hide Example Installation Output**

          The following shows an example of the output of the above installation step.

          .. code-block::

             <FIXME: add installation output here>

   .. group-tab:: Linux - Debian

      In a terminal, run

      .. code-block:: bash

          sudo apt-get install make gcc g++ python3 python3-dev python3-pip

>           sudo apt-get install swig  # optional, needed for one of the examples

Reviewer 4: ```suggestion
```

      .. container:: toggle

          .. container:: header

              **Show/Hide Example Installation Output**

          The following shows an example of the output of the above installation step.

          .. code-block::

             <FIXME: add installation output here>

   .. group-tab:: Linux - Red Hat

      In a terminal, run

      .. code-block:: bash

          sudo yum install make gcc gcc-c++ libstdc++-devel python3 python3-devel python3-pip

>           sudo yum install swig  # optional, needed for one of the examples

Reviewer 4: ```suggestion
```

      .. container:: toggle

          .. container:: header

              **Show/Hide Example Installation Output**

          The following shows an example of the output of the above installation step.

          .. code-block::

>              <FIXME: add installation output here>

Reviewer 2: Not sure this is all that useful

Author: This has been suggested by @imphil in https://github.com/cocotb/cocotb/pull/1637#issuecomment-613958024 and I like the idea - we don't need to show the full output if it's long, but in order to get a feel for what a successful step looks like, it's nice and reassuring for newbies.

Reviewer 2: I suppose it's not harmful. In the interest of getting this merged soon though, I'd be tempted to merge without this, and add it in a follow-up.

Author: Ok, removed.

.. _install:

************
Installation
************

> Using cocotb requires :ref:`installation of pre-requisites<install-pre-requisites>` and

Reviewer 4: ```suggestion
Using cocotb requires :ref:`installation of prerequisites<install-prerequisites>` and
```

Author: Unfortunately, since I an ``.. include::``ing the file where this anchor used to live several times for the different OSes, we cannot really link to it anymore, so I removed it now.
I changed the spelling as suggested.
